### PR TITLE
[TTAHUB-1380] Create caseload and filter recipient search by it

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Add a panel to the account management page -> "my groups".
This lets you create/edit/delete groups for usage on the recipient search page, as a filter called "groups"

## How to test
Try creating, editing, deleting groups. Apply as a filter and ensure the results are correct.

Note: There is a known issue that we are NOT addressing as part of MVP, which is that when using the My Groups selector as a user with access to all regions, it is nearly impossible to select a grant (because your list will have something like 1700 options, and this creates roughly one trillion DOM nodes). Since this was intended to help specialists with their caseloads, we are comfortable addressing this post-MVP. 

## Issue(s)
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1380

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/components/filter/constants.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/* eslint-disable import/prefer-default-export */
+export const handleArrayQuery = (q) => {
```
Comment (AdamAdHocTeam) [inline frontend/src/components/filter/constants.js:2]: This probably belongs in a helper file as apposed to a constants file.

File: src/middleware/checkIdParamMiddleware.js [hunk 2]
```diff
@@ -124,3 +124,18 @@ export function checkObjectiveTemplateIdParam(req, res, next) {
   auditLogger.error(msg);
```
Comment (AdamAdHocTeam) [inline src/middleware/checkIdParamMiddleware.js:124]: Is this part of the change for groups?

File: frontend/src/pages/RecipientSearch/index.js [hunk 3]
```diff
@@ -5,16 +5,17 @@ import React, {
 import PropTypes from 'prop-types';
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/RecipientSearch/index.js:5]: Is this part of the change for groups?

File: src/routes/groups/handlers.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    res.json(groupResponse);
+  } catch (e) {
+    auditLogger.error(`${NAMESPACE} getGroup ${e}`);
+    res.sendStatus(httpCodes.INTERNAL_SERVER_ERROR);
+  }
+}
+
+export async function createGroup(req: Request, res: Response) {
+  try {
+    const userId = await currentUserId(req, res);
```
Comment (AdamAdHocTeam) [inline src/routes/groups/handlers.ts:48]: Should we be performing some sort of authorization here? Or maybe its in the calling function.

File: src/scopes/index.js [hunk 5]
```diff
@@ -4,6 +4,7 @@ import { _ } from 'lodash';
 import { activityReportsFiltersToScopes as activityReport } from './activityReport';
 import { grantsFiltersToScopes as grant } from './grants';
 import { goalsFiltersToScopes as goal } from './goals';
+// import { recipientsFiltersToScopes as recipient } from './recipients';
```
Comment (AdamAdHocTeam) [inline src/scopes/index.js:7]: Not sure if you want this commented/removed.

**Review by AdamAdHocTeam — COMMENTED** [review]
Overall it looks good and works from the UI.

Some things:
1. The recipient selector seems "laggy" to me (could be my pc).
2. The pr "seems" to contains changes to other files.
3. I'm not clear how case load fits into this (might be the next steps tho).

**Comment by thewatermethod** [comment]
> The recipient selector seems "laggy" to me (could be my pc).

Adam and I talked about this on a slack call - this is the known issue where a user with all regions will load too many grants for this selector to be usable. This was discussed previously at a standup and we are ok with launching with that being the case

**Comment by thewatermethod** [comment]
@AdamAdHocTeam think I've addressed your concerns if you want to give it the old approve

**Review by GarrettEHill — APPROVED** [review]

**Pull request merged**